### PR TITLE
Refactor: Relax Int64 type restriction to accept any integer dtype

### PR DESCRIPTION
**Pull request by atulcthakur**
# ALCHEMI Toolkit Pull Request

## Description

 `nvalchemi` enforces `Int64` for `AtomicNumbers`, `EdgeIndex`, and `BatchIndices`, which rejects valid `int32` tensors at validation time. This is unnecessary — `nvalchemiops` is entirely int32-based (Warp kernels are hard-coded to `wp.int32`), and PyTorch ops like indexing, `scatter_add_`, and `bincount` all work fine with `int32`. The current code does `int64→int32→int64` round-trip conversions on every neighbor list call, wasting memory and compute.

 This PR relaxes the three type aliases from `Int64` to Integer (which accepts any integer dtype) and removes the unnecessary `.long()` calls scattered across the codebase. MACE wrapper is untouched since it may require `int64`.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [ ] Documentation update
- [x] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Changes Made

  - `nvalchemi/_typing.py` — Int64 → Integer for AtomicNumbers, EdgeIndex, BatchIndices; removed unused Int64 import
  - `nvalchemi/data/batch.py` — removed .long() from batch, edge_ptr, num_nodes_per_graph, num_edges_per_graph, and index_select
  - `nvalchemi/models/_ops/neighbor_filter.py` — removed .long() from neighbor matrix clamping, batch_idx cell lookup, and neighbor list
  i/j index extraction
  - `nvalchemi/models/lj.py` — removed .long() from scatter_add_ batch_idx
  - `nvalchemi/models/ewald.py `— removed .long() from scatter_add_ batch_idx
  - `nvalchemi/models/pme.py` — removed .long() from scatter_add_ batch_idx
  - `nvalchemi/dynamics/hooks/neighbor_list.py` — removed .long() from edge source indexing, batch assignment, and staging buffer expansion
  - `nvalchemi/dynamics/integrators/npt.py` — removed .long() from bincount batch
  - `nvalchemi/dynamics/integrators/nph.py` — removed .long() from bincount batch
  - `nvalchemi/dynamics/integrators/nvt_nose_hoover.py` — removed .long() from bincount batch
  - `nvalchemi/dynamics/optimizers/fire.py` — removed .long() from bincount batch
  - `nvalchemi/dynamics/_ops/nose_hoover.py` — removed .long() from bincount batch_idx
  - `test/data/test_atomic_data.py` — 3 new tests: int32 atomic_numbers accepted, int32 edge_index accepted, int64 regression guard
## Testing

- [x] Unit tests pass locally (`make pytest`)
- [x] Linting passes (`make lint`)
- [x] New tests added for new functionality meets coverage expectations?

## Checklist

- [x] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [x] I have performed a self-review of my code
- [ ] I have added docstrings to new functions/classes
- [ ] I have updated the documentation (if applicable)

## Additional Notes

> [!TIP]
> This repository uses Greptile, an AI code review service, to help conduct
> pull request reviews. We encourage contributors to read and consider suggestions
> made by Greptile, but note that human maintainers will provide the necessary
> reviews for merging: Greptile's comments are **not** a qualitative judgement
> of your code, nor is it an indication that the PR will be accepted/rejected.
> We encourage the use of emoji reactions to Greptile comments, depending on
> their usefulness and accuracy.

### CODE REVIEW

File: test/data/test_atomic_data.py [hunk 1]
```diff
@@ -82,6 +82,35 @@ def test_optional_system_fields(self):
...
+        """AtomicData should accept int32 edge_index."""
+        data = AtomicData(
+            positions=torch.randn(3, 3),
+            atomic_numbers=torch.ones(3, dtype=torch.long),
+            edge_index=torch.zeros(4, 2, dtype=torch.int32),
+        )
+        assert data.edge_index.dtype == torch.int32
+        assert data.edge_index.shape == (4, 2)
+
+    def test_int64_still_works(self):
```
Comment (laserkelvin) [inline test/data/test_atomic_data.py]: I would just parametrize as a single test

**Review by laserkelvin — CHANGES_REQUESTED** [review]
I'm fine with these changes - my only concern is downstream, whether or not certain kernels used in dynamics actually do expect certain integer types for things like `batch_idx`.

If you can find one or two unit test cases with neighbor hooks and dynamics and parametrize them to use either int32 or int64 for these tensors, that will tell us if we're okay I think

File: test/data/test_atomic_data.py (hunk 1 above)
Comment (atulcthakur) [reply test/data/test_atomic_data.py]: Done

File: nvalchemi/dynamics/hooks/neighbor_list.py [hunk 2]
```diff
@@ -444,8 +444,8 @@ def _update_edges_group(
 
         from nvalchemi.data.level_storage import SegmentedLevelStorage
 
-        src_atoms = edge_index[:, 0].long()  # (E,)
-        graph_per_edge = batch.batch.long()[src_atoms]  # (E,)
+        src_atoms = edge_index[:, 0]  # (E,)
+        graph_per_edge = batch.batch[src_atoms]  # (E,)
         seg_lengths = torch.bincount(graph_per_edge, minlength=B).to(torch.int32)
```
[the remark this answers is not in the document]
Comment (atulcthakur) [reply nvalchemi/dynamics/hooks/neighbor_list.py:454]: Incorrect. `bincount` signature accepts any int tensor.

File: nvalchemi/data/batch.py [hunk 3]
```diff
@@ -171,7 +171,7 @@ def batch(self) -> Tensor:
         atoms = self._atoms_group
         if atoms is None:
             return torch.tensor([], dtype=torch.long, device=self.device)
-        return atoms.batch_idx.long()
+        return atoms.batch_idx
```
[the remark this answers is not in the document]
Comment (atulcthakur) [reply nvalchemi/data/batch.py:175]: Is this about Batch being empty? Can this ever occur?
